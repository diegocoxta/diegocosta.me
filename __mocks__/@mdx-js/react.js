module.exports = {
  MDXProvider: jest.fn(({ children }) => children),
};
