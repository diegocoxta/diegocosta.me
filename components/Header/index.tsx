import Link from 'next/link';

import styles from './styles.module.css';

export interface HeaderProps {
  name: string;
  href?: string;
}

export default function Header({
  href = '/',
  name,
  children,
}: React.PropsWithChildren<HeaderProps>): React.ReactElement {
  const [firstName, lastName] = name.split(' ');

  return (
    <header className={styles.container}>
      <Link className={styles.link} href={href} data-testid="logo-link">
        <h1 className={styles.name}>
          {firstName}
          {lastName && (
            <span className={styles.lastName} data-testid="logo-lastname">
              {lastName[0]}.
            </span>
          )}
        </h1>
      </Link>
      <div className={styles.navBar}>{children}</div>
    </header>
  );
}
