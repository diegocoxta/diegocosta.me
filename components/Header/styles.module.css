.container {
  composes: container from '~/components/Container/styles.module.css';
  margin: 16px auto 40px;
  display: flex;
  justify-content: space-between;
  align-items: center;

  @media (width >= 760px) {
    margin: 40px auto;
  }
}

.link {
  text-decoration: none;
  box-shadow: none;
  display: flex;
  align-items: center;

  &:focus {
    outline: none;
  }
}

.name {
  color: var(--main-title-color);
  font-weight: 700;
  text-transform: lowercase;
  margin: 0;
  font-size: 15vw;

  @media (width >= 760px) {
    font-size: 80px;
  }
}

.lastName {
  color: var(--main-accent-color);
}

.navBar {
  display: flex;
  justify-content: space-between;
  align-items: center;
}
