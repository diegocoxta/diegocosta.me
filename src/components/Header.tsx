import React from 'react';
import styled from 'styled-components';
import { KBarProvider, KBarProviderProps } from 'kbar';

import BaseContainer from './Container';
import Logo, { LogoProps } from './Logo';
import Navigation from './Navigation';
import ThemeSwitcher from './ThemeSwitcher';

const Container = styled(BaseContainer)`
  margin: 16px auto 40px auto;
  display: flex;
  justify-content: space-between;
  align-items: center;

  @media (min-width: 760px) {
    margin: 40px auto 40px auto;
  }
`;

const NavBar = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
`;

export interface HeaderProps {
  name: LogoProps['name'];
  actions: KBarProviderProps['actions'];
}

export default function Header(props: HeaderProps): React.ReactElement {
  return (
    <Container>
      <Logo name={props.name} size="large" />
      <NavBar>
        <ThemeSwitcher />
        <KBarProvider actions={props.actions}>
          <Navigation />
        </KBarProvider>
      </NavBar>
    </Container>
  );
}
