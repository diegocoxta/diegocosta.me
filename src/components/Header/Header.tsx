import React from 'react';
import styled, { keyframes } from 'styled-components';

import { Link } from '~/utils/i18n';
import FixedContainer from '~/components/FixedContainer';
import ThemeSwitcher from '~/components/ThemeSwitcher';

const Content = styled.header`
  margin: 16px 0 40px 0;
  display: flex;
  justify-content: space-between;
  align-items: center;

  @media (min-width: 760px) {
    margin: 40px 0 40px 0;
  }
`;

const avatarAnimation = keyframes`
  0%, 100% { border-radius: 63% 37% 54% 46%/55% 48% 52% 45% }

  14% { border-radius: 40% 60% 54% 46%/49% 60% 40% 51% }

  28% { border-radius: 54% 46% 38% 62%/49% 70% 30% 51% }

  42% { border-radius: 61% 39% 55% 45%/61% 38% 62% 39% }

  56% { border-radius: 61% 39% 67% 33%/70% 50% 50% 30% }

  70% { border-radius: 50% 50% 34% 66%/56% 68% 32% 44% }

  84% { border-radius: 46% 54% 50% 50%/35% 61% 39% 65% }
`;

const Avatar = styled.img`
  animation: ${avatarAnimation} 10s linear infinite alternate forwards;
  width: 15vw;
  margin-right: 30px;
  height: 100%;

  @media (min-width: 760px) {
    width: 90px;
  }
`;

const Name = styled.h1`
  font-size: 15vw;
  color: ${({ theme }) => theme.titleColor};
  font-weight: 700;
  text-transform: lowercase;
  margin: 0;

  @media (min-width: 760px) {
    font-size: 100px;
  }
`;

const LastName = styled.span`
  color: ${({ theme }) => theme.accentColor};
`;

const StyledLink = styled(Link)`
  text-decoration: none;
  box-shadow: none;
  display: flex;
  align-items: center;

  :focus {
    outline: none;
  }
`;

const Options = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
`;

interface HeaderProps {
  author: string;
  avatar?: string;
}

export default function Header(props: HeaderProps): React.ReactElement {
  const { avatar, author } = props;
  const [name, lastname] = author.split(' ');

  return (
    <FixedContainer>
      <Content>
        <StyledLink to="/">
          {avatar && <Avatar src={avatar} />}
          <Name>
            {name}
            {lastname && <LastName data-testid="header-lastname">{lastname[0]}.</LastName>}
          </Name>
        </StyledLink>
        <Options>
          <ThemeSwitcher />
        </Options>
      </Content>
    </FixedContainer>
  );
}
