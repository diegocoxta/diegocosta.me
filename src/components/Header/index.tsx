import React from 'react';
import { useStaticQuery, graphql } from 'gatsby';

import Header from './Header';

export default (): React.ReactElement => {
  const {
    site: {
      siteMetadata: { author, avatar },
    },
  } = useStaticQuery(
    graphql`
      query HeaderComponent {
        site {
          siteMetadata {
            author
            avatar
          }
        }
      }
    `
  );

  return <Header author={author} avatar={avatar} />;
};
