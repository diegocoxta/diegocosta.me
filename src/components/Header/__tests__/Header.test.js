import React from 'react';
import { render } from '@testing-library/react';

import Header from '../Header';

jest.mock('../components/ThemeSwitcher', () => () => <p>ThemeSwitcher</p>);
jest.mock('../components/LanguageSwitcher', () => () => <p>LanguageSwitcher</p>);
jest.mock('../components/Navigation', () => () => <p>Navigation</p>);
jest.mock('../components/Metatags', () => () => <p>Metatags</p>);

describe('<Header />', () => {
  it('renders properly', () => {
    const { baseElement, getByTestId, getByText } = render(
      <Header author="John Doe" description={{ en: 'You can hear more about here:' }} />
    );
    expect(getByTestId('header-lastname')).toBeTruthy();
    expect(getByText('John')).toBeTruthy();
    expect(baseElement).toMatchSnapshot();
  });

  it('does not render the last name', () => {
    const { baseElement, queryByTestId } = render(<Header author="John" />);
    expect(queryByTestId('header-lastname')).toBeFalsy();
    expect(baseElement).toMatchSnapshot();
  });

  it('renders the navigation properly', () => {
    const { baseElement, getByTestId, queryAllByTestId, getAllByTestId, getByText } = render(
      <Header
        author="John Doe"
        description="You can hear more about here:"
        navigation={[
          { label: '', url: '' },
          { label: '', url: '' },
        ]}
      />
    );

    expect(getByTestId('header-lastname')).toBeTruthy();
    expect(getByText('John')).toBeTruthy();
    expect(getByTestId('about-me-navigation-list')).toBeTruthy();
    expect(queryAllByTestId('about-me-navigation-item')).toBeTruthy();
    expect(getAllByTestId('about-me-navigation-item').length).toEqual(2);
    expect(baseElement).toMatchSnapshot();
  });
});
