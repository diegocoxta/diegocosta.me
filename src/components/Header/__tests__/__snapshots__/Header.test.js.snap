// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> does not render the last name 1`] = `
.c0 {
  max-width: 800px;
  margin: 0 auto;
  padding: 0 20px;
}

.c1 {
  margin: 16px 0 40px 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  font-size: 15vw;
  font-weight: 700;
  text-transform: lowercase;
  margin: 0;
}

.c2 {
  -webkit-text-decoration: none;
  text-decoration: none;
  box-shadow: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2:focus {
  outline: none;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

@media (min-width:760px) {
  .c1 {
    margin: 40px 0 40px 0;
  }
}

@media (min-width:760px) {
  .c3 {
    font-size: 100px;
  }
}

<body>
  <div>
    <section
      class="c0"
    >
      <main>
        <header
          class="c1"
        >
          <a
            class="c2"
            href="/"
          >
            <h1
              class="c3"
            >
              John
            </h1>
          </a>
          <div
            class="c4"
          >
            <p>
              ThemeSwitcher
            </p>
          </div>
        </header>
      </main>
    </section>
  </div>
</body>
`;

exports[`<Header /> renders properly 1`] = `
.c0 {
  max-width: 800px;
  margin: 0 auto;
  padding: 0 20px;
}

.c1 {
  margin: 16px 0 40px 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  font-size: 15vw;
  font-weight: 700;
  text-transform: lowercase;
  margin: 0;
}

.c2 {
  -webkit-text-decoration: none;
  text-decoration: none;
  box-shadow: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2:focus {
  outline: none;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

@media (min-width:760px) {
  .c1 {
    margin: 40px 0 40px 0;
  }
}

@media (min-width:760px) {
  .c3 {
    font-size: 100px;
  }
}

<body>
  <div>
    <section
      class="c0"
    >
      <main>
        <header
          class="c1"
        >
          <a
            class="c2"
            href="/"
          >
            <h1
              class="c3"
            >
              John
              <span
                class=""
                data-testid="header-lastname"
              >
                D
                .
              </span>
            </h1>
          </a>
          <div
            class="c4"
          >
            <p>
              ThemeSwitcher
            </p>
          </div>
        </header>
      </main>
    </section>
  </div>
</body>
`;
